pkg/clustermanager/provision_node.go

```diff
@@ -10,24 +9,23 @@ import (
 
 const maxErrors = 3
 
-// K8sVersion is the version that will be used to install kubernetes
-var K8sVersion = flag.String("k8s-version", "1.16.2-00", "The version of the k8s debian packages that will be used during provisioning")
-
 // NodeProvisioner provisions all basic packages to install docker, kubernetes and wireguard
 type NodeProvisioner struct {
-	clusterName  string
-	node         Node
-	communicator NodeCommunicator
-	eventService EventService
+	clusterName       string
+	node              Node
+	communicator      NodeCommunicator
+	eventService      EventService
+	kubernetesVersion string
 }
 
 // NewNodeProvisioner creates a NodeProvisioner instance
-func NewNodeProvisioner(clusterName string, node Node, communicator NodeCommunicator, eventService EventService) *NodeProvisioner {
+func NewNodeProvisioner(clusterName string, node Node, communicator NodeCommunicator, eventService EventService, kubernetesVersion string) *NodeProvisioner {
```

Comment:
Maybe we should give the manager object as argument here?

Reply:
How would this look like?

Reply:
I made a commit to do this